modules/apps/object/object-dynamic-data-mapping-storage/src/main/java/com/liferay/object/dynamic/data/mapping/storage/internal/ObjectDDMStorageAdapter.java

```diff
@@ -215,12 +212,16 @@ private long _getObjectDefinitionId(Long ddmStructureId)
 
 			DDMFormField ddmFormField = ddmFormValue.getDDMFormField();
 
+			if (!objectFieldNames.contains(ddmFormField.getFieldReference())) {
+				continue;
+			}
+
 			Value value = ddmFormValue.getValue();
 
 			Map<Locale, String> values = value.getValues();
 
 			properties.put(
-				StringUtil.toLowerCase(ddmFormField.getName()),
+				StringUtil.toLowerCase(ddmFormField.getFieldReference()),
```

Comment:
@gabrielwas, why did you decide to use the field reference instead of the field name? Is it because the user can change/view it?

Reply:
Because the user can change the fieldReference and match it with the object field name, did you get it?

Reply:
Yeah! Thanks. :slightly_smiling_face: